gammapy/catalog/hawc.py

```diff
@@ -0,0 +1,174 @@
+            info = 'basic,position,spectrum'
+
+        ss = ''
+        ops = info.split(',')
+        if 'basic' in ops:
+            ss += self._info_basic()
+        if 'position' in ops:
+            ss += self._info_position()
+        if 'spectrum' in ops:
+            ss += self._info_spectrum()
+
+        return ss
+
+    def _info_basic(self):
+        """Print basic info."""
+        d = self.data
+        ss = '\n*** Basic info ***\n\n'
+        ss += 'Catalog row index (zero-based) : {}\n'.format(d['catalog_row_index'])
+        ss += '{:<15s} : {}\n'.format('Source name:', d['source_name'])
+
+        return ss
+
+    def _info_position(self):
+        """Print position info."""
+        d = self.data
+        ss = '\n*** Position info ***\n\n'
+
+        ss += 'Measurement:\n'
+        ss += '{:20s} : {:.3f}\n'.format('RA', d['ra'])
+        ss += '{:20s} : {:.3f}\n'.format('DEC', d['dec'])
+        ss += '{:20s} : {:.3f}\n'.format('GLON', d['glon'])
+        ss += '{:20s} : {:.3f}\n'.format('GLAT', d['glat'])
+        ss += '{:20s} : {:.3f}\n'.format('Position error', d['pos_err'])
+
+        return ss
+
+    def _info_spectrum(self):
+        """Print spectral info."""
+        d = self.data
+        ss = '\n*** Spectral info ***\n\n'
+
+        ss += 'Spectrum 1:\n'
+        ss += '{:20s} : {:.3} +- {:.3} {}\n'.format('Flux at 7 TeV', d['spec0_dnde'].value, \
+            d['spec0_dnde_err'].value, 'cm-2 s-1 TeV-1')
+        ss += '{:20s} : {:.3f} +- {:.3f}\n'.format('Spectral index', d['spec0_index'], d['spec0_index_err'])
+        ss += '{:20s} : {:1}\n\n'.format('Test radius', d['spec0_radius'])
+
+        if(np.isnan(d['spec1_dnde'])):
+            ss += 'No second spectrum available for this source'
+        else:
+            ss += 'Spectrum 2:\n'
+            ss += '{:20s} : {:.3} +- {:.3} {}\n'.format('Flux at 7 TeV', d['spec1_dnde'].value, \
+                d['spec1_dnde_err'].value, 'cm-2 s-1 TeV-1')
+            ss += '{:20s} : {:.3f} +- {:.3f}\n'.format('Spectral index', d['spec1_index'], d['spec1_index_err'])
+            ss += '{:20s} : {:1}'.format('Test radius', d['spec1_radius'])
+
+        return ss
+
+    # TODO: Add an @property?
+    def _n_spec(self):
+        d = self.data
+        N = 0
+        if(np.isnan(d['spec1_dnde'])):
+            N = 1
+        else:
+            N = 2
+
+        return N
+
+    @property
+    def spectral_model(self):
+        """Source spectral model (`~gammapy.spectrum.models.SpectralModel`).
+        """
+
+        data = self.data
+
+        if(self._n_spec() == 1):
+            pars, errs = {}, {}
+            model_class = PowerLaw
+            pars['amplitude'] = data['spec0_dnde']
+            errs['amplitude'] = data['spec0_dnde_err']
+            pars['index'] = data['spec0_index']*u.Unit('')
+            errs['index'] = data['spec0_index_err']*u.Unit('')
+            pars['reference'] = 7*u.TeV #TODO: Check whether 7Tev is correct by definition of 
```

Comment:
I think it has to be 7TeV because all fluxes in the catalog are at 7TeV. But I don't know exactly how this pars['reference'] is handled in line 120 and whether the model is correct afterwards.

Reply:
Yes, 7 TeV is correct here. Remove the TODO.